examples/sphere/shallow_water.jl

```diff
@@ -421,172 +398,251 @@ function rhs!(dYdt, y, parameters, t)
+        (
+            test_name == "mountain" ? MountainTest(α) :
+            (
+                test_name == "rossby_haurwitz" ? RossbyHaurwitzTest(α) :
+                (
+                    test_name == "barotropic_instability" ?
+                    BarotropicInstabilityTest(α) : SteadyStateTest(α)
+                )
+            )
         )
-        # Height errors over time
-        relL1err = Array{Float64}(undef, div(T, dt))
-        for t in 1:div(T, dt)
-            relL1err[t] =
-                norm(solution[t].h .- Y0_global.h, 1) / norm(Y0_global.h, 1)
+    # Set up discretization
+    ne = 9 # the rossby_haurwitz test case's initial state has a singularity at the pole. We avoid it by using odd number of elements
+    Nq = 4
+
+    domain = Domains.SphereDomain(test.params.R)
+    mesh = Meshes.EquiangularCubedSphere(domain, ne)
+    quad = Spaces.Quadratures.GLL{Nq}()
+    if usempi
+        grid_topology = Topologies.DistributedTopology2D(context, mesh)
+        global_grid_topology = Topologies.Topology2D(mesh)
+        space = Spaces.SpectralElementSpace2D(grid_topology, quad)
+        global_space = Spaces.SpectralElementSpace2D(global_grid_topology, quad)
+    else
+        grid_topology = Topologies.Topology2D(mesh)
+        global_space =
+            space = Spaces.SpectralElementSpace2D(grid_topology, quad)
+    end
+    coords = Fields.coordinate_field(space)
+    f = set_coriolis_parameter(space, test)
+    h_s = surface_topography(space, test)
+    Y = set_initial_condition(space, test)
+    if !usempi
+        Y0_global = deepcopy(Y)
+    else
+        Y0_global_values = DataLayouts.gather(context, Fields.field_values(Y))
+        if ClimaComms.iamroot(context)
+            Y0_global = Fields.Field(Y0_global_values, global_space)
         end
-        Plots.png(
-            Plots.plot(
-                [1:dt:T],
-                relL1err,
-                xlabel = "time [s]",
-                ylabel = "Relative L₁ err",
-                label = "",
-            ),
-            joinpath(path, "HeightRelL1errorVstime.png"),
-        )
-        linkfig(
-            relpath(
-                joinpath(path, "HeightRelL1errorVstime.png"),
-                joinpath(@__DIR__, "../.."),
-            ),
-            "Height relative L1 error over time",
-        )
+    end
+    ghost_buffer = usempi ? Spaces.create_ghost_buffer(Y) : nothing
+    # Set up RHS function
+    dYdt = similar(Y)
+    parameters =
+        (; f = f, h_s = h_s, ghost_buffer = ghost_buffer, params = test.params)
+    rhs!(dYdt, Y, parameters, 0.0)
+
+    # Solve the ODE
+    dt = 9 * 60
+    T = 86400 * 2
+
+    prob = ODEProblem(rhs!, Y, (0.0, T), parameters)
+    integrator = OrdinaryDiffEq.init(
+        prob,
+        SSPRK33(),
+        dt = dt,
+        saveat = dt,
+        progress = true,
+        adaptive = false,
+        progress_message = (dt, u, p, t) -> t,
+    )
```

Comment:
The allocations script expects that `integrator` is defined at global scope, can we split this function up into
```julia
integrator = get_integrator(...)
if haskey(ENV, "CI_PERF_SKIP_RUN") # for performance analysis
    throw(:exit_profile)
end
solve_problem(integrator, ...)
```
?